packages/swing-store/src/swingStore.js

```diff
@@ -63,6 +63,8 @@ import { doRepairMetadata } from './repairMetadata.js';
  * @typedef {{
  *   dump: (includeHistorical?: boolean) => SwingStoreDebugDump,
  *   serialize: () => Buffer,
+ *   setEnableTranscriptRolloverCompression: (mode: boolean) => void,
```

Comment:
The problem with making it an option on store creation is that for the test to work I need to flip the toggle midway through the test, because the test is making sure that things still work right when you have uncompressed spans in a store that had data put into it before the compression feature was added.  Also, since this is a test-only feature, I wouldn't want it to be available on the non-test API, of which the options bag in question is a part.  However, assuming I did it right, the toggle that's there is strictly part of the test/debug API that won't be exposed to normal code.

Reply:
Would a more realistic replication would be to reopen the store in the new compressing mode, making the construction option viable ?

Reply:
That would complicate the test considerably, due to the need for it to then run in two passes, one to generate the pre-compression store and the other to exercise the post-compression-enabled behavior.  I'm unconvinced that there's any payoff at all to the extra effort.

Reply:
Resolution: leave the toggle as is for now, but leave a comment describing our desire to eventually do away with it.